### PR TITLE
Adapt popup height to contents

### LINKED ISSUE #1931: Popup: proper height/width handling with reasonable defaults
## Feature Description
It would be good to limit popup to some width/height by default and allow overriding these values per use case.

If the popup content exceeds the limits, scrolling will help.

Additionally, the popup should adjust to width/height changes of the content. For example, if the popup is displayed in the bottom edge of the viewport and the content was previously 100px, and then changes to 300px, part of the popup will get hidden "below" the bottom edge of the viewport.[ Video with example](https://github.com/google/web-stories-wp/pull/1743#discussion_r427682509).

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## QA Instructions

See the testing instructions in https://github.com/google/web-stories-wp/pull/2306

### LINKED ISSUE #2343: Tooltips: tooltip with longer text goes out of the screen
## Bug Description
When longer text is added to the tooltip, it stays on one line causing the tooltip to go out of the screen.

## Expected Behaviour
Tooltip should wrap the text instead to always stay within the borders of the viewport.

## Steps to Reproduce
1. Add a tooltip with longer text to something in the Design Panel
2. See the tooltip partially not visible.

## Screenshots
Screenshot 2020-06-10 at 17 16 20

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by wassgha**
## Summary

Makes the popup component dynamically adapt to height changes of its content, thereby preventing scrollable overflow (especially visible on the calendar element which switching views, see https://github.com/google/web-stories-wp/pull/1743#discussion_r427682509). 

## To-do

- [x] Trigger the height change automatically without need for a scroll event

## User-facing changes

- Link popups are now fixed (no longer showing an unnecessary scroll bar)
- Calendar will now be fully visible without need for scrolling

## Testing Instructions
I
1. Open the calendar.
2. Scroll up (so that the calendar would be in the bottom of the viewport, resize the window if needed to achieve that)
3. Now click on the month (May 2020) to open the month view - this will change the content height.
4. Verify that the calendar/popup adjusts the size and all parts of the calendar content are still visible.

II
1. Add a video element.
2. Resize it to be very narrow.
3. Now hover over it: verify the Play button is visible outside (just aside) of the Video element.
---

Fixes #1931 , see context on https://github.com/google/web-stories-wp/pull/1743#discussion_r427682509
Part of #1082
Fixes #2343

### CODE REVIEW

File: assets/src/edit-story/components/inspector/document/publish/publishTime.js [hunk 1]
```diff
@@ -140,14 +141,18 @@ function PublishTime() {
...
-          onChange={handleDateChange}
-          is12Hour={use12HourFormat}
-          forwardedRef={dateTimeNode}
-        />
+        {({ propagateDimensionChange }) => (
+          <DateTime
+            value={date}
+            onChange={(value, close = false) => {
+              handleDateChange(value, close);
+              propagateDimensionChange();
```
Comment (wassgha) [inline assets/src/edit-story/components/inspector/document/publish/publishTime.js]: @miina this is currently not triggered when switching between views, which makes it require a scroll before the change is in effect. I tried using `onViewChange` from the `react-calendar` but that did not work either, any ideas?

**Comment by wassgha** [comment]
@miina any input on this please?

**Comment by miina** [comment]
@googlebot I consent.

**Comment by miina** [comment]
@wassgha: 
`onViewChange` seems to work when testing locally -- added a commit with that (de1b502) -- could you confirm that it's not working for you?

File: assets/src/edit-story/components/inspector/document/publish/publishTime.js [hunk 2]
```diff
@@ -140,14 +141,19 @@ function PublishTime() {
...
-          onChange={handleDateChange}
-          is12Hour={use12HourFormat}
-          forwardedRef={dateTimeNode}
-        />
+        {({ propagateDimensionChange }) => (
+          <DateTime
+            value={date}
+            onChange={(value, close = false) => {
+              handleDateChange(value, close);
+
```
Comment (miina) [inline assets/src/edit-story/components/inspector/document/publish/publishTime.js]: (Whoops, left an empty line in, didn't run the commit through the linter, sorry)

File: assets/src/edit-story/components/form/dateTime/datePicker.js [hunk 3]
```diff
@@ -43,13 +41,18 @@ function DatePicker({ currentDate, onChange }) {
...
+        onChange={handleOnChange}
+        onViewChange={onViewChange}
+      />
     </CalendarWrapper>
   );
 }
 
 DatePicker.propTypes = {
   onChange: PropTypes.func.isRequired,
+  onViewChange: PropTypes.func.isRequired,
```
Comment (dvoytenko) [inline assets/src/edit-story/components/form/dateTime/datePicker.js]: IMHO this should be an optional arg

File: assets/src/edit-story/components/popup/index.js [hunk 4]
```diff
@@ -157,9 +153,10 @@ function Popup({
           fillWidth={fillWidth}
           fillHeight={fillHeight}
           placement={placement}
-          maxHeight={maxHeight}
         >
-          {children}
+          {typeof children === 'function'
```
Comment (dvoytenko) [inline assets/src/edit-story/components/popup/index.js]: This is pretty complicated. Can we just use a context here? Or even simply bubble a custom event and handle it here? Also, would a container with the `ResizeObserver` work for this?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

**Comment by wassgha** [comment]
Ready!

File: assets/src/edit-story/components/form/dateTime/datePicker.js (hunk 3 above)
Comment (miina) [reply assets/src/edit-story/components/form/dateTime/datePicker.js]: @wassgha This still seems to be required though, did you leave it in purposefully?

Comment (wassgha) [reply assets/src/edit-story/components/form/dateTime/datePicker.js]: No, sorry, it got reverted :)

**Review by miina — APPROVED** [review]
LGTM!

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**